File: app/src/main/java/com/weberbox/pifire/utils/log/DebugLogTree.java
package com.weberbox.pifire.utils.log;

import android.os.Build;
import android.util.Log;

import androidx.annotation.NonNull;

import timber.log.Timber;

public class DebugLogTree extends Timber.DebugTree {

    private final String prefix;

    public DebugLogTree(String prefix) {
        this.prefix = prefix;
    }

    @Override
    protected void log(int priority, String tag, @NonNull String message, Throwable t) {
        // Workaround for devices that do not show lower priority logs
        if (Build.MANUFACTURER.equals("HUAWEI") || Build.MANUFACTURER.equals("samsung")) {
            if (priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO)
                priority = Log.ERROR;
        }
        super.log(priority, prefix + "." + tag, message, t);
    }

    @Override
    protected String createStackElementTag(StackTraceElement element) {
        return String.format("(%s:%s) - %s",
                element.getFileName(),
                element.getLineNumber(),
                element.getMethodName());
    }
}

File: app/src/main/java/com/weberbox/pifire/core/util/DebugLogTree.kt
package com.weberbox.pifire.core.util

import android.os.Build
import android.util.Log
import timber.log.Timber

class DebugLogTree(private val prefix: String) : Timber.DebugTree() {

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        // Workaround for devices that do not show lower priority logs
        var updatedPriority = priority
        if (Build.MANUFACTURER == "HUAWEI" || Build.MANUFACTURER == "samsung") {
            if (priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO) {
                updatedPriority = Log.ERROR
            }
        }
        super.log(updatedPriority, "$prefix.$tag", message, t)
    }

    override fun createStackElementTag(element: StackTraceElement): String {
        return String.format(
            "(%s:%s) - %s",
            element.fileName,
            element.lineNumber,
            element.methodName
        )
    }
}